projects/packages/jetpack-mu-wpcom/src/features/launchpad/class-launchpad-task-lists.php

```diff
@@ -424,14 +424,33 @@ private function load_calypso_path( $task ) {
 			return null;
 		}
 
-		// Require that the string start with `/`, but don't allow `//`.
-		if ( '/' !== substr( $calypso_path, 0, 1 ) || '/' === substr( $calypso_path, 1, 1 ) ) {
+		if ( ! $this->is_valid_url_or_path( $calypso_path ) ) {
 			return null;
 		}
 
 		return $calypso_path;
 	}
 
+	/**
+	 * Checks if a string is a valid URL or path.
+	 *
+	 * @param string $input The string to check.
+	 * @return boolean
+	 */
+	private function is_valid_url_or_path( $input ) {
```

Comment:
Minor thought: to be extra helpful to our future selves, it may be worth using an even more explicit function name to indicate that we allow admin URLs and absolute paths?
```suggestion
	/**
	 * Checks if a string is a valid admin URL or an absolute path.
	 *
	 * @param string $input The string to check.
	 * @return boolean
	 */
	private function is_valid_admin_url_or_absolute_path( $input ) {
```